### DISCUSSION TITLE
Allow Expression Trees with async/await

### DISCUSSION THREAD

**Discussion by jmaine** [dotnet/csharplang#1505] [Language Ideas]
Example:
`Expression<Func<Func<Task<int>> Task<bool>>> lambda = async f => await f() < 5;  `

'Await' here will be translated to System.Linq.Expressions.AwaitExpression without creating a state machine when c# compiler runs.

lambda.compile() would be responsible translating the expression tree with AwaitExpression into the needed state machines that make the newly compiled async/await method work.

**Comment by Joe4evr** [comment]
Related/duplicate of #158.
